apps/frontend/src/lib/components/Chat.svelte

```diff
@@ -0,0 +1,226 @@
+		onChatStart: () => void;
+		suggestions?: Array<{
+			title: string;
+			description: string;
+			suggestedText: string;
+		}>;
+	}
+
+	let { client, assistantId, threadId, userName, userEmail, chatStarted, onChatStart, suggestions }: Props = $props();
+
+	const defaultSuggestions = [
+		{
+			title: "Creative Brainstorming",
+			description: "Generate ideas for projects, writing, or problem-solving",
+			suggestedText: "Help me brainstorm ideas for a creative project"
+		},
+		{
+			title: "Writing Assistance",
+			description: "Draft, edit, or improve emails, documents, and more",
+			suggestedText: "Help me write and improve some text"
+		},
+		{
+			title: "Learn Something New",
+			description: "Get clear explanations on topics you're curious about",
+			suggestedText: "Explain a complex topic in simple terms"
+		},
+		{
+			title: "Problem Solving",
+			description: "Break down challenges and find solutions together",
+			suggestedText: "Help me analyze and solve a problem"
+		}
+	];
+
+	const displaySuggestions = suggestions || defaultSuggestions;
+
+	let current_input = $state('');
+	let is_streaming = $state(false);
+	let messages = $state<Array<Message>>([{ type: 'ai', text: 'How can I help you?' }]);
+
+	async function inputSubmit() {
+		if (current_input) {
+			if (!chatStarted) {
+				onChatStart();
+			}
+			
+			messages.push({
+				type: 'user',
+				text: current_input
+			});
+
+			is_streaming = true;
+
+			let aimessage: Message = {
+				type: 'ai',
+				text: ''
+			};
+
+			messages.push(aimessage);
+
+			for await (const chunk of streamAnswer(client, threadId, assistantId, current_input)) {
+				console.log('Processing chunk in inputSubmit:', chunk);
+				if (chunk.type === 'tool') {
+					const toolMsg: ToolMessageType = {
+						type: 'tool',
+						text: "I'm using tools...",
+						tool_name: chunk.tool_name,
+						payload: chunk.tool_payload,
+						collapsed: true,
+					};
+					messages.push(toolMsg);
+					
+					aimessage = {
+						type: 'ai',
+						text: ''
+					};
+					messages.push(aimessage);
+					messages = [...messages];
+				} else if (chunk.type === 'text') {
+					aimessage.text += chunk.text;
+					
+					let aiMessageIndex = -1;
+					for (let i = messages.length - 1; i >= 0; i--) {
+						if (messages[i].type === 'ai') {
+							aiMessageIndex = i;
+							break;
+						}
+					}
+					
+					if (aiMessageIndex !== -1) {
+						messages[aiMessageIndex] = { ...aimessage };
+						messages = [...messages];
+					}
+				}
+			}
+
+			current_input = '';
+			is_streaming = false;
+		}
+	}
+
+	function scrollToMe(node: HTMLElement) {
+		node.scrollIntoView({ behavior: 'smooth', block: 'center' });
+		return {
+			destroy() {}
+		};
+	}
+
+	function getGreeting() {
+		if (userName) {
+			return `Hey ${userName}, how can I help you today?`;
+		} else if (userEmail) {
+			return `Hey ${userEmail.split('@')[0]}, how can I help you today?`;
+		} else {
+			return 'How can I help you today?';
+		}
```

Comment:
I think it's clearer to just pass a single `userName` and let the caller determine what it considers the user's name.